packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts

```diff
@@ -124,7 +124,11 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
     this.loadBalancer = props.loadBalancer;
     this.protocol = protocol;
     this.certificateArns = [];
-    this.certificateArns.push(...(props.certificateArns || []));
+
+    // Attach certificates
+    if (props.certificateArns) {
```

Comment:
This does not preclude it from being an empty list, just means it's not `undefined`.

Reply:
Added a check for an empty list as well.